docs/src/arrays.md

```diff
@@ -0,0 +1,440 @@
+```
+
+### Other useful identities
+
+These identities are particularly useful:
+
+```math
+\begin{align*}
+d\left( \operatorname{real}(A) \right) &= \operatorname{real}(dA)\\
+d\left( \operatorname{conj}(A) \right) &= \operatorname{conj}(dA)\\
+d\left( A^T \right) &= dA^T\\
+d\left( A^H \right) &= dA^H\\
+d\left( \sum_{j}  A_{i \ldots j \ldots k} \right) &=
+        \sum_{j} dA_{i \ldots j \ldots k},
+\end{align*}
+```
+
+where $\cdot^H$ is the conjugate transpose (the `adjoint` function).
+
+## Deriving reverse-mode rules
+
+Reverse-mode rules are a little more involved.
+For a real scalar function $s = g(C)$, the differential of $s$ is the real part of the inner product (`LinearAlgebra.dot`) of the adjoint of $C$ (i.e. $\overline{C}$), and the differential of $C$:
+
+```math
+ds = \operatorname{real}\left( \langle \overline{C}, dC \rangle \right)
+   = \operatorname{real}\left(
+         \sum_{i,\dots,j} \operatorname{conj}(\overline{C}_{i,\dots,j}) ~dC_{i,\dots,j}
+     \right),
+```
+
+where $\operatorname{conj}(\cdot)$ is the complex conjugate (`conj`), and $\operatorname{real}(\cdot)$ is the real part of its argument (`real`).
+
+For matrices and vectors, we can write this as
+
+$$ds = \operatorname{real}(\operatorname{tr}(\overline{C}^H dC)),$$
+
+where $\operatorname{tr}$ is the matrix trace (`LinearAlgebra.tr`) function.
+
+We can write the corresponding expression for $dA$ and $dB$ as
+
+```math
+\begin{align*}
+ds &= \operatorname{real}(\operatorname{tr}(\overline{C}^H dC)) &&\\
+   &= \operatorname{real}\left( \operatorname{tr}\left(
+          \overline{C}^H \frac{\partial f}{\partial A} dA +
+          \overline{C}^H \frac{\partial f}{\partial B} dB
+      \right) \right) && \text{substitute } dC \text{ from } \eqref{cdiff}\\
+   &= \operatorname{real}\left( \operatorname{tr}\left(
+          \overline{C}^H \frac{\partial f}{\partial A} dA
+      \right) \right) +
+      \operatorname{real}\left( \operatorname{tr}\left(
+          \overline{C}^H \frac{\partial f}{\partial B} dB
+      \right) \right) && \text{expand using } \eqref{trexpand}
+\end{align*}
+```
+
+By applying the same definition of $ds$ to the intermediates $A$ and $B$, we get
+
+```math
+ds = \operatorname{real}\left( \operatorname{tr}(\overline{A}^H  dA) \right) +
+     \operatorname{real}\left( \operatorname{tr}(\overline{B}^H ~dB) \right)
+```
+
+Combining these two identities and solving for $\overline{A}$ and $\overline{B}$ gives us
+
+```math
+\begin{align*}
+    \overline{A} &= (\overline{A}^H)^H
+                  = \left( \overline{C}^H \frac{\partial f}{\partial A} \right)^H
+                  = \left( \frac{\partial f}{\partial A} \right)^H \overline{C}\\
+    \overline{B} &= \left( \frac{\partial f}{\partial B} \right)^H \overline{C}
+\end{align*}
+```
+
+Giles' method for deriving pullback functions is to first derive the differential identity (i.e. pushforward) using the above approach, then pre-multiply by $\overline{C}^H$, and take the real trace.
```

Comment:
```suggestion
Giles's method for deriving pullback functions is to first derive the differential identity (i.e. pushforward) using the above approach, then pre-multiply by $\overline{C}^H$, and take the real trace.
```

Reply:
I wonder if this paragraph and the properties of the trace function should come before the example in which they are used? Or perhaps for this little section to have a sub-heading? It feels a bit lost amongst the text right now.